src/deck_assembler.py

```diff
@@ -0,0 +1,65 @@
+import hashlib
+from pathlib import Path
+import genanki
+from parsed_codeblock import ParsedCodeBlock
+
+class DeckAssembler:
+    def __init__(self):
+        pass
+    
+    def assemble_deck_from_codeblocks(self, codeblocks: list[ParsedCodeBlock], output_path: Path):
+        
+        decks = {}
+        
+        for codeblock in codeblocks:
+            note_name = self.get_attribute_from_codeblock_headers(codeblock.headers, "name")
+        
+            note_deck_name = self.get_attribute_from_codeblock_headers(codeblock.headers, "deck")
+            
+            assert note_name is not None
+            assert note_deck_name is not None
+            
+            # TODO document on github how note ids are generated from names
+            note = genanki.Note(
+                guid=quick_hash(note_name + "_" + note_deck_name),
+                model=genanki.CLOZE_MODEL,
+                fields=[codeblock.content, '']
+            )
+           
+            if note_deck_name in decks:
+                decks[note_deck_name].add_note(note)
+            else:
+                # TODO document on github how deck ids are generated from deck name
+                deck = genanki.Deck(
+                    deck_id=quick_hash(note_deck_name),
+                    name=note_deck_name
+                )
+
+                deck.add_note(note)
+                
+                print(f"Added {note_name} to deck {note_deck_name}")
+                
+                decks[note_deck_name] = deck
+        
+        package_name = "deck_package.apkg" 
+        
+        package_path = output_path / package_name 
+         
+        genanki.Package(decks.values()).write_to_file(package_path.absolute())
```

Comment:
`write_to_file` does not belong here https://github.com/Abb4/codeblock-to-anki/issues/7